pkg/operator/operators/klusterlet/controllers/klusterletcontroller/klusterlet_controller.go

```diff
@@ -272,45 +274,30 @@ func (n *klusterletController) sync(ctx context.Context, controllerContext facto
 		}
 	}
 
-	appliedCondition := meta.FindStatusCondition(klusterlet.Status.Conditions, klusterletApplied)
+	klusterlet.Status.ObservedGeneration = klusterlet.Generation
+
 	if len(errs) == 0 {
-		appliedCondition = &metav1.Condition{
+		meta.SetStatusCondition(&klusterlet.Status.Conditions, metav1.Condition{
 			Type: klusterletApplied, Status: metav1.ConditionTrue, Reason: "KlusterletApplied",
-			Message: "Klusterlet Component Applied"}
+			Message: "Klusterlet Component Applied"})
 	} else {
-		if appliedCondition == nil {
-			appliedCondition = &metav1.Condition{
+		if meta.FindStatusCondition(klusterlet.Status.Conditions, klusterletApplied) == nil {
```

Comment:
it seems this check is not necessary `if meta.FindStatusCondition(klusterlet.Status.Conditions, klusterletApplied) == nil {`?
Think about if a klusterlet upgraded to a new version and applied failed, the old version's klusterletApplied exists, the new version applied failed error will not be recorded.